Add instagram reels functionality, cell tiling efficiency, and stop creating windows once the monitor is full

Jurassic001 opened the pull request:
Alternate between YouTube shorts and Instagram reels, tile cells efficiently across the entire monitor, and create windows up until N = num_windows or the windows are going off the screen.

Jurassic001 committed:
Add instagram reels functionality, cell tiling efficiency, and stop creating windows once the monitor is full

Jurassic001 committed:
Run clippy

Jurassic001 committed:
Run clippy

Jurassic001 commented:
Hold on, git is having a hard time keeping track of two whole forks

Jurassic001 committed:
Run clippy and commit the changes to the right remote this time

Rational1ty reviewed (changes requested):
Few things:
- FYI you can just cargo run instead of build/start
- Rename `num_screens` -> `num_windows`
- Use separate `YT_NAV_H` and `INSTA_NAV_H` even if same value
- If statements can use pattern matching assignment like `let (url, w, h) = if ...`
- Avoid creating loop variables outside of the loop and avoid making them mutable

Jurassic001 committed:
Make requested changes

Attempted to avoid as many mutable loop variables outside of the loop as possible, but no approach I could think of worked to calculate window X position while factoring in window overflowing

what am I doing with my life.

Rational1ty reviewed (changes requested):
Just a few small changes left

Jurassic001 committed:
Make requested changes
